build/tasks/apply-backstage-annotation.yaml

```diff
@@ -0,0 +1,107 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: apply-backstage-annotation
+spec:
+  description: |
+    Retrieves the OCI manifest for the RHTAP plugins, generates a Base64 hash from the contents in index.json 
+    and applies the annotation 'io.backstage.dynamic-packages' with the value being the hash.
+  params:
+    - name: IMAGE_URL
+      type: string
+      description: The URL of the image to extract index.json from
+    - name: IMAGE_DIGEST
+      type: string
+      description: The digest of the image
+  workspaces:
+    - name: manifest-data
+      description: Workspace for passing the new manifest.json file
+  steps:
+    - name: generate-backstage-annotation
+      image: registry.access.redhat.com/ubi8/ubi-minimal:latest
+      script: |
+        #!/bin/bash
+        set -e
+  
+        # Change to "$(params.IMAGE_URL)@$(params.IMAGE_DIGEST)"
+        TEST_IMAGE=quay.io/redhat-user-workloads/rhtap-shared-team-tenant/rhtap-backstage-plugins:on-pr-969d69a71f6f903bd16171b4d46330242f7b4a0d@sha256:df89d0bd76b606add3c419838e64c6c58d76448807b96ff70f59e2e66698c530
+        # Used for fetching manifest
+        # Change to "$(params.IMAGE_URL)"
+        TEST_IMAGE_NO_SHA=quay.io/redhat-user-workloads/rhtap-shared-team-tenant/rhtap-backstage-plugins:on-pr-969d69a71f6f903bd16171b4d46330242f7b4a0d
+
+        # Install tools
+        echo "Installing tools..."
+        microdnf install -y tar findutils curl jq gzip
+
+        # Create directory for extracted files
+        mkdir -p /tmp/extracted
+
+        # Install Oras
+        echo "Installing ORAS..."
+        # Docs: https://oras.land/docs/installation
+        VERSION="1.2.2"
+        curl -LO "https://github.com/oras-project/oras/releases/download/v${VERSION}/oras_${VERSION}_linux_amd64.tar.gz"
```

Comment:
Could you into maybe split the into more tasks by using already pre-built images from a tssc pipeline like you do on next step by using `konflux-ci/oras`.

Reply:
Yeah this can be shrunk down using other images. I think the only dependency I need is `tar` and `gzip` to get the container files.